### ISSUE TITLE
[bug] AssertionError with prompt_logprobs and logits_processors both set

### ISSUE THREAD

**Issue by gx16377** [vllm-project/vllm#2800]
version 0.3.0
with prompt_logprobs and logits_processors both set
```
def proc(token_ids, logits_row):
    return logits_row
sampling_params = SamplingParams(temperature=0.8, top_p=0.95, prompt_logprobs=100, logits_processors=[proc])
```
got error
```
  File "/opt/conda/lib/python3.10/site-packages/vllm/model_executor/layers/sampler.py", line 74, in forward
    logits = _apply_logits_processors(logits, sampling_metadata)
  File "/opt/conda/lib/python3.10/site-packages/vllm/model_executor/layers/sampler.py", line 162, in _apply_logits_processors
    assert logits_row_idx == logits.shape[0]
AssertionError
```

**Comment by gx16377** [comment]
In fact, I want a way to get logits (tensor, instead of those dicts of floats or so) of all tokens (including prompts) to calculate ppl outside vllm, tried these params and got the error

**Comment by huyiwen** [comment]
Same issue. I think the author @noamgat @hongxiayang @simon-mo might know the answer.

**Comment by huyiwen** [comment]
API added here: #1469

**Comment by huyiwen** [comment]
I believe this issue is related to `prompt_logprobs`. A quick solution would be to set `prompt_logprobs` to None (if it applies in your case).

**Comment by DarkLight1337** [comment]
Closing as completed by #3899.
